sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -1107,11 +1107,6 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
       }
     }
 
-    // Note: Before altering table partitions in Hive, you *must* set the current database
-    // to the one that contains the table of interest. Otherwise you will end up with the
-    // most helpful error message ever: "Unable to alter partition. alter is not possible."
-    // See HIVE-2742 for more detail.
-    client.setCurrentDatabase(db)
```

Comment:
Sorry, I meant `a special case`.

Reply:
This does not change the behavior. Just to restore the original current database in the hive client. I do not think the current implementation has any issue, because we do not rely on the current database in the hive client. However, I am also fine to restore the original. This change does not have any harm but just make the codes more stable for the future refactoring.